src/backend/cuda/subconfigure.m4

```diff
@@ -89,7 +91,30 @@ if test "$with_cuda" != "no" ; then
         ],[
             have_cuda=no
             AC_MSG_RESULT([no])
+
+            # if using the Intel icc compiler retry without the '-ccbin' option and fallback to
+            # binary compatible gcc
+            if test "${ac_save_CC}" == "icc"; then
+                NVCC_FLAGS="$save_NVCC_FLAGS -ccbin gcc"
```

Comment:
I am thinking that we can just try the default (remove the `-ccbin` option) and see if it works. I would rather not to maintain the extra logic of checking `icc` and picking `gcc`.

Reply:
Okay. I was just thinking that we can ensure binary compatibility  this way (with the C code of the non-CUDA files compiled using `icc`). Or don't you see an issue there?

Reply:
How about when the check failed here we simply spit a "helpful" message and ask users to set `NVCC` and re-configure? We did this in `test/mpi/configure` but neglected to do the same for yaksa -- https://github.com/pmodels/mpich/pull/5968

Reply:
Mh aren't we having the exact the same issue [here](https://github.com/pmodels/mpich/blob/main/test/mpi/configure.ac#L736) and even omit the test during configure time? I would assume (did not try it out yet) that the compile would fail when `icc` is used as host compiler.

Probably it is a matter of taste. Do we wann to provide a fallback (and additionally print a warning in such cases). Or do we rather want the user to explicitly set the host compiler if the currently chosen compiler conflicts with `nvcc`?

Reply:
I think we should always have a mechanism to allow users to apply their own settings. So only do the check when `NVCC` is not set by the user. Since users can fallback on that, I think we can just fail for those corner cases such as `icc`.

Yes, we should test the `ccbin` at least once in configure.

Reply:
Okay, I changed it accordingly. Users may now explicitly  set `NVCC` (including flags/options). The retry has been removed.

Reply:
What do you think? Is this was you intended?